### ISSUE TITLE
MD025/single-title/single-h1

### ISSUE THREAD

**Issue by AliceonFire** [DavidAnson/markdownlint#199]
The **MD025** rule complains about *"Multiple top-level headings in the same document"* seemingly because there is a title in the article's YAML frontmatter along with an h1 heading in the same document. I can't think of why else the warning would appear. Is this intentional?

```
---
title: "How Titles and Headings Affect SEO | SEOMasters"
---

# Does Google Weigh Title and H1 Equally?

## the answer may surprise you…
```

Single opinions such as **MD026/no-trailing-punctuation** can be disregarded easily enough, but it seems MD025 could be useful with separation of the single-title and single-h1 rules. Would it be possible to toggle this decoupling in a setting rather than turning the rule off entirely?

**Comment by DavidAnson** [comment]
This is deliberate (for the reason you suggest) and is customizable. The rule documentation explains more: https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md025---multiple-top-level-headings-in-the-same-document

**Comment by AliceonFire** [comment]
The **MD025** rule documentation states `To disable the use of front matter by this rule, specify "" for front_matter_title.`

This is not seen as a valid editor configuration setting:
```json
{
    "[markdown]":{
        "front_matter_title": ""
    }
}
```

Placing it within a markdown file's YAML frontmatter also has no effect:
```markdown
---
front_matter_title: ""
title: "Page Title"
---

# Page Header
```

Where is the specification of "front_matter_title" to be carried out?

**Comment by DavidAnson** [comment]
This part of the documentation outlines what the configuration object should look like: https://github.com/DavidAnson/markdownlint#optionsconfig

Rule parameters are nested; here is a specific example for your scenario: https://github.com/DavidAnson/markdownlint/blob/6f7c0aac131f21dd41a2a26810779c993521b733/test/hugo-quickstart-example-clean.json

Just in case you are using the VS Code extension, here's what things looks like in the configuration for that editor: https://github.com/DavidAnson/vscode-markdownlint#markdownlintconfig
